### PR TITLE
Bugfix KNOWAGE-8044

### PR DESCRIPTION
**Pull request by BojanSovticEngIT**
Possible bugfix https://production.eng.it/jira/browse/KNOWAGE-8044

Files changed:

- knowagewhatifengine/src/main/webapp/js/src/angular_1.x/scripts/directives/components/TableToolbarRight.js ([216, 255]),
- knowagecockpitengine/src/main/webapp/js/src/angular_1.4/cockpit/services/cockpitModule_backwardCompatibility.js ([144]),
- knowagecockpitengine/src/main/webapp/js/src/angular_1.4/cockpit/services/cockpitModule_datasetServices.js ([1044]),
- knowagecockpitengine/src/main/webapp/js/src/angular_1.4/cockpit/services/cockpitModule_exportWidgetService.js ([89]),
- knowagecockpitengine/src/main/webapp/js/src/angular_1.4/cockpit/services/cockpitModule_variableService.js ([88])

### CODE REVIEW

File: knowagecockpitengine/src/main/webapp/js/src/angular_1.4/cockpit/services/cockpitModule_backwardCompatibility.js [hunk 1]
```diff
@@ -146,7 +146,7 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 						for(var k in model.content.columnSelectedOfDataset){
 							if(model.content.columnSelectedOfDataset[k].isCalculated) {
 								model.content.columnSelectedOfDataset[k].aggregationSelected = "NONE";
-								if(!model.content.columnSelectedOfDataset[k].formula.match(/(AVG|SUM|MIN|MAX|COUNT|COUNT DISTINCT|TOTAL_SUM|TOTAL_AVG|TOTAL_MIN|TOTAL_MAX|TOTAL_COUNT)\(\"+[\D\+\-\*\/\w]*\)/gm)){
+								if(!model.content.columnSelectedOfDataset[k].formula.match(/(AVG|SUM|MIN|MAX|COUNT|COUNT DISTINCT|TOTAL_SUM|TOTAL_AVG|TOTAL_MIN|TOTAL_MAX|TOTAL_COUNT)\("[\D+\-*/\w]*"\)/gm)){
```
Comment (davide-zerbetto) [inline knowagecockpitengine/src/main/webapp/js/src/angular_1.4/cockpit/services/cockpitModule_backwardCompatibility.js:149]: Reg expression logic is not the same as before, please check it

File: knowagecockpitengine/src/main/webapp/js/src/angular_1.4/cockpit/services/cockpitModule_datasetServices.js [hunk 2]
```diff
@@ -1041,7 +1041,7 @@ $mdPanel,cockpitModule_widgetSelection,cockpitModule_properties,cockpitModule_ut
 	this.replaceStringVariables = function (obj){
 	var objString = angular.copy(obj);
 	objString = JSON.stringify(objString);
-    objString = objString.replace(/\$V\{([a-zA-Z0-9\-\_]*){1}(?:.([a-zA-Z0-9\-\_]*){1})?\}/g,function(match,p1,p2){
+    objString = objString.replace(/\$V\{([a-zA-Z0-9_]+)(?::[a-zA-Z0-9_]+)?}/g,function(match,p1,p2){
```
Comment (davide-zerbetto) [inline knowagecockpitengine/src/main/webapp/js/src/angular_1.4/cockpit/services/cockpitModule_datasetServices.js:1044]: Reg expression was changed: "-" has been removed and why adding ":"?

File: knowagecockpitengine/src/main/webapp/js/src/angular_1.4/cockpit/services/cockpitModule_exportWidgetService.js [hunk 3]
```diff
@@ -86,7 +86,7 @@
 	var replaceStringVariables = function (obj){
 		var objString = angular.copy(obj);
 		objString = JSON.stringify(objString);
-  		  objString = objString.replace(/\$V\{([a-zA-Z0-9\-\_]*){1}(?:.([a-zA-Z0-9\-\_]*){1})?\}/g,function(match,p1,p2){
+  		  objString = objString.replace(/\$V\{([a-zA-Z0-9_]+)(?::[a-zA-Z0-9_]+)?}/g,function(match,p1,p2){
```
Comment (davide-zerbetto) [inline knowagecockpitengine/src/main/webapp/js/src/angular_1.4/cockpit/services/cockpitModule_exportWidgetService.js:89]: Same as previous comment

File: knowagecockpitengine/src/main/webapp/js/src/angular_1.4/cockpit/services/cockpitModule_variableService.js [hunk 4]
```diff
@@ -85,7 +85,7 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 		this.getVariablePlaceholders = function(textToParse){
 			var tempText = angular.copy(textToParse);
 			if(tempText){
-				tempText = tempText.replace(/\$V\{([a-zA-Z0-9\-\_]*){1}(?:.([a-zA-Z0-9\-\_]*){1})?\}/g,function(match,p1,p2){
+				tempText = tempText.replace(/\$V\{[a-zA-Z0-9_]+(?::[a-zA-Z0-9_]+)?\}/g,function(match,p1,p2){
```
Comment (davide-zerbetto) [inline knowagecockpitengine/src/main/webapp/js/src/angular_1.4/cockpit/services/cockpitModule_variableService.js:88]: Same as previous comment

File: knowagewhatifengine/src/main/webapp/js/src/angular_1.x/scripts/directives/components/TableToolbarRight.js [hunk 5]
```diff
@@ -212,7 +212,7 @@ function tableToolobarController($scope, $timeout, $window, $mdDialog, $http, $s
 	//Getting all buttons from XML and moving them inside grids(arrays) where they need to be
 	filterXMLResult = function(res) {
 
-		var regEx = /([A-Z]+_*)+/g;
+		var regEx = /([A-Z]+(_[A-Z]+)*[A-Z]*)+/g;
```
Comment (davide-zerbetto) [inline knowagewhatifengine/src/main/webapp/js/src/angular_1.x/scripts/directives/components/TableToolbarRight.js:215]: Maybe it shoud be /([A-Z_]+)+/g;

File: knowagewhatifengine/src/main/webapp/js/src/angular_1.x/scripts/directives/components/TableToolbarRight.js [hunk 6]
```diff
@@ -251,7 +251,7 @@ function tableToolobarController($scope, $timeout, $window, $mdDialog, $http, $s
 
 	//Getting clicked buttons trough XML (clicked="true")
 	filterClickedButtons = function(data){
-		var regEx = /([A-Z]+_*)+/g;
+		var regEx = /([A-Z]+(_[A-Z]+)*[A-Z]*)+/g;
```
Comment (davide-zerbetto) [inline knowagewhatifengine/src/main/webapp/js/src/angular_1.x/scripts/directives/components/TableToolbarRight.js:254]: Maybe it shoud be /([A-Z_]+)+/g;

**Review by davide-zerbetto — CHANGES_REQUESTED** [review]

**Comment by BojanSovticEngIT** [comment]
Updated

**Comment by BojanSovticEngIT** [comment]
Updated

**Pull request merged**